src/generated/todo/getTodosSaga.js

```diff
@@ -0,0 +1,75 @@
+// This file is generated
+
+import { takeEvery } from 'redux-saga';
+import { call, put } from 'redux-saga/effects';
+
+// FIXME - implement proper api
+function api(aPathParam, aQueryParam) {
+  return new Promise((resolve) => {
+    // Use superagent
```

Comment:
I'd use `superagent` patched with a promise library to get proper Promise APIs. 

Sample patch using `superagent-promise-plugin` and `es6-promise`
```js
import Promise from 'es6-promise';
import superagentPromisePlugin from 'superagent-promise-plugin';
import request from 'superagent';

Promise.polyfill();
export default superagentPromisePlugin.patch(request);
```
The patched version can be consumed anywhere else in the app like so
```js
import r from './superagent-patch';

export const fetchTodos = () => r.get('/todos').then(r => r.body.data).catch(e => e);
```